app/gateway/routing/expressions.md

```diff
@@ -65,106 +63,30 @@ This predicate example has the following structure:
+Routes are inserted into each router with the appropriate `priority` field set. The priority is a positive integer that defines the order of evaluation of the router. The bigger the priority, the sooner a Route will be evaluated. In the case of duplicate priority values between two Routes in the same router, their order of evaluation is undefined. The router is
+updated incrementally as configured Routes change.
 
-```
-http.path ~ r#"/foo/bar/\d+"#
-```
-
-### Case insensitive path matching
-
-If you want to ignore case when performing the path match, use the `lower()` modifier on the field
-to ensure it always returns a lowercase value:
-
-```
-lower(http.path) == "/foo/bar"
-```
-
-This will match requests with a path of `/foo/bar` and `/FOO/bAr`, for example.
-
-### Match by header value
-
-If you want to match incoming requests by the value of header `X-Foo`, do the following:
-
-```
-http.headers.x_foo ~ r#"bar\d"#
-```
+As soon as a Route yields a match, the router stops matching and the matched Route is used to process the current request/connection.
 
-If there are multiple header values for `X-Foo` and the client sends more than
-one `X-Foo` header with different value, the above example will ensure **each** instance of the
-value will match the regex `r#"bar\d"#`. This is called "all" style matching, meaning each instance
-of the field value must pass the comparison for the predicate to return `true`. This is the default behavior.
-
-If you do not want this behavior, you can turn on "any" style of matching which returns
-`true` for the predicate as soon as any of the values pass the comparison:
-
-```
-any(http.headers.x_foo) ~ r#"bar\d"#
-```
-
-This will return `true` as soon as any value of `http.headers.x_foo` matches regex `r#"bar\d"#`.
-
-Different transformations can be chained together. The following is also a valid use case
-that performs case-insensitive matching:
-
-```
-any(lower(http.headers.x_foo)) ~ r#"bar\d"#
-```
-
-### Regex captures
-
-You can define regex capture groups in any regex operation which will be made available
-later for plugins to use. Currently, this is only supported with the `http.path` field:
-
-```
-http.path ~ r#"/foo/(?P<component>.+)"#
-```
-
-The matched value of `component` will be made available later to plugins such as
-[Request Transformer Advanced](https://docs.konghq.com/hub/kong-inc/request-transformer-advanced/how-to/templates/).
-
-## Expression router examples (TCP, TLS, UDP)
-
-### Match by source IP and destination port
-
-```
-net.src.ip in 192.168.1.0/24 && net.dst.port == 8080
-```
-
-This matches all clients in the `192.168.1.0/24` subnet and the destination port (which is listened to by Kong)
-is `8080`. IPv6 addresses are also supported.
-
-### Match by SNI (for TLS routes)
-
-```
-tls.sni =^ ".example.com"
-```
+![Router matching flow](https://docs.konghq.com/assets/images/products/gateway/reference/expressions-language/router-matching-flow.png)
```

Comment:
For reviewers: I wrote this down to convert to Mermaid, but I'm just doing a 1:1 copy pretty much and wanted to get this reviewed. I'm going to need to learn how to escape some characters in Mermaid, so this might take me a bit.